### PR TITLE
Booster badge / Notification (EXPOSUREAPP-9193, 9199, 9202)

### PR DESCRIPTION
**Pull request by mtwalli**
### Changes 
- Show booster badge once user is eligible 
- Show home screen notification if user has not been notified before
- Display rule description in Vaccine status card
- Add Booster entry in Test menu 

### Testing 
1- Switch to MOCK_CLOUD environment 
2- Scan vaccination certificates generated by `cwa vc gen -s v.0.dt=-150days` 

> 📝 NOTE : You can try different value than `150` be creative 😃  

3- Go to booster test menu -> Hit run booster rules 
4- Notification should appear in system tray 
5- Bottom Navigation Bar should display a badge count `=` number of eligible persons  

### TODO
- [x] Unit testing

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt [hunk 1]
```diff
@@ -101,8 +101,7 @@ private val DccValidationRule.Type.asExternalType: Type
     get() = when (this) {
         DccValidationRule.Type.ACCEPTANCE -> Type.ACCEPTANCE
         DccValidationRule.Type.INVALIDATION -> Type.INVALIDATION
-        DccValidationRule.Type.BOOSTER_NOTIFICATION ->
-            throw IllegalArgumentException("Not yet supported by dgca verifier")
+        DccValidationRule.Type.BOOSTER_NOTIFICATION -> Type.ACCEPTANCE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt:104]: TODO clarify this

**Comment by SamuraiKek** [comment]
First time I tested, I received the system notification but the badge count didn't show up in the bottom nav. 
After that I tried to refresh booster rules, clear booster rules and run booster rules again but nothing happened. 
I deleted the certificate and scanned it again then ran booster rules again but still nothing happened.
Then I erased app data, scanned the QR again and ran booster rules but still nothing happened. 
I also reinstalled the app for good measure but still nothing happened.

Am I missing something?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/core/business/wrapper/Mappings.kt:104]: @BMItr  clarified with Thomas, this part seems to be missing for Android , it is in iOS part 
https://github.com/corona-warn-app/cwa-app-tech-spec/blob/894f9517ee41889bae64f365494e74c60d3f7252/docs/spec/dgc-overview-client.md#ios-invoking-certlogic-for-booster-notifications

**Comment by mtwalli** [comment]
> I also reinstalled the app for good measure but still nothing happened.

What does the logs say ?

**Comment by SamuraiKek** [comment]
> What does the logs say ?

Update: Had a look over this with @mtwalli  and it looks like it's the cli tool. The certificates are generated randomly and not all of them have the booster rules applied to them. If you get this issue while testing, just keep generating certificates until you get one of them with booster rules (you'll know it when you click "Run Booster Rules" and receive the system notification)

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Review by SamuraiKek — APPROVED** [review]
LGTM. Seems to work nicely when the certs behave 😄

**Review by AlexanderAlferov — APPROVED** [review]
Looks great 🚀

**Review by BMItr — APPROVED** [review]
image
👌

**Pull request merged**